### ISSUE TITLE
Better differentiable interpolation

### ISSUE THREAD

**Issue by crowsonkb** [lucidrains/big-sleep#24]
I notice that on line 179 of `big_sleep.py` you use `F.interpolate()` to downsample: https://github.com/lucidrains/big-sleep/blob/main/big_sleep/big_sleep.py#L179

I've observed that `F.interpolate()` is bad at downsampling because it doesn't prefilter its input and this leaves weird blocky artifacts in the gradient of the original-sized image. Is that why you are doing 128 random-sized cutouts? I wrote a prefiltering resampling function today that uses `F.interpolate()` for upsampling but which low-pass filters its input (with a Lanczos 3 kernel it computes) before calling it for downsampling. It's at https://gist.github.com/crowsonkb/a905773ba4d7aa5cd7671315e464369c. Hope this helps!

**Comment by lucidrains** [comment]
@crowsonkb merged in! thank you Katherine!
